lib/models/application.js

```diff
@@ -1302,21 +1261,21 @@ const getApplicationModel = function (deps, opts) {
 		 * 	if (error) throw error;
 		 * });
 		 */
-		revokeSupportAccess: (nameOrSlugOrId) =>
-			getId(nameOrSlugOrId)
-				.then((applicationId) =>
-					pine.patch({
-						resource: 'application',
-						id: applicationId,
-						body: { is_accessible_by_support_until__date: null },
-					}),
-				)
-				.catch((err) => {
-					if (isNotFoundResponse(err)) {
-						treatAsMissingApplication(nameOrSlugOrId, err);
-					}
-					throw err;
-				}),
+		revokeSupportAccess: async (nameOrSlugOrId) => {
+			try {
+				const applicationId = await getId(nameOrSlugOrId);
+				return pine.patch({
```

Comment:
```suggestion
				return await pine.patch({
```
Otherwise the catch will not run if `pine.patch` fails, which is more of an issue if `nameOrSlugOrId` is a number, in which case `getId` doesn't check with the API.